foodbornenyc/sources/yelp_fast.py

```diff
@@ -336,162 +336,52 @@ def upsert_yelpfile_to_db(filename, geocode=True):
                     # print biz_count
                     continue
 
-            # create the Location object
-            loc = dict(biz['location'])                     
-            street_address = xstr(loc['address'][0])\
-                +', ' +xstr(loc['address'][1])\
-                +', ' +xstr(loc['address'][2])\
-                +', '+xstr(loc['city']) +', '+xstr(loc['state'])\
-                +' '+xstr(loc['postal_code'])
-            street_address = street_address.lower()
-
-            # haven't encountered this location so make one
-            if street_address not in db_locations and street_address not in unloaded_locations.keys():
-                try: # some don't have lat,longs
-                    location = {
-                                'street_address':street_address,
-                                'latitude':float(loc['coordinate']['latitude']),
-                                'longitude':float(loc['coordinate']['longitude']),
-                                'line1':xstr(loc['address'][0]),
-                                'line2':xstr(loc['address'][1]),
-                                'line3':xstr(loc['address'][2]),
-                                'city':xstr(loc['city']),
-                                'country':xstr(loc['country']),
-                                'postal_code':xstr(loc['postal_code']),
-                                'state':xstr(loc['state'])
-                                }
-                except TypeError: # so they can't get converted to float
-                    # so reverse geocode them (if enabled)
+            # make note of new Locations
+            location = location_dict_yelp(biz['location'])
```

Comment:
This is the only place that might be less efficient: we construct a dictionary for each location instead of first computing the key and then making a dictionary if the key is a new location's.